src/AutoRest.CSharp.V3/AutoRest/Plugins/ClientOptionsWriter.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using AutoRest.CSharp.V3.Generation.Writers;
+using AutoRest.CSharp.V3.Output.Models.Types;
+using Azure.Core;
+
+namespace AutoRest.CSharp.V3.AutoRest.Plugins
+{
+    internal class ClientOptionsWriter
+    {
+        public static void WriteClientOptions(CodeWriter writer, BuildContext context)
```

Comment:
What is this for?

Reply:
Generates ***ClientOptions classes that are required for public ctors. https://github.com/Azure/autorest.csharp/pull/644/files#diff-1facde7a0e058c6acf8cc086f0635dc8

Reply:
Got it. All clients we generate use the same **ClientOptions, so it makes sense to generate it with the name of the library, and not the name of the client itself. For example, it isn't UsagesClientOptions, but instead StorageManagementClientOptions for any client in StorageManagement.